cumulus_library/studies/core/core_templates/documentreference.sql.jinja

```diff
@@ -0,0 +1,64 @@
+{% import 'core_utils.jinja' as utils %}
+CREATE TABLE core__documentreference AS
+WITH temp_documentreference AS (
+    SELECT DISTINCT
+        {{- utils.basic_cols(
```

Comment:
How useful is `basic_cols` vs just using `nullable_cols`? Early detection of oddities, I guess. But like, what if in the data there **is** a null `id` field (that would be messed up, but bear with me). How should our pipeline treat that? I believe ETL requires the `id` field to exist, but doesn't verify that the value itself isn't null. And then this would error out when they tried to build core?

Just curious if we couldn't just treat everything as nullable and simplify some templating. With our growing collection of data-sanity studies (`quality` and the other one you made I don't remember the name), I wonder if `core` is the right place to enforce basic stuff like this, rather than being graceful about data it receives, which the user probably couldn't easily fix anyway.

Reply:
i think the only advantage is query speed, and i don't know how big that is.